File: Chap1-Events/1-4EventPropagationDemo/childlineedit.cpp
#include "childlineedit.h"
#include <QDebug>
#include <QKeyEvent>

ChildLineEdit::ChildLineEdit(QWidget *parent) : ParentLineEdit(parent)
{

}

void ChildLineEdit::keyPressEvent(QKeyEvent *event)
{
    //    qDebug() << "ChildLineEdit keyPressEvent";
    //    ParentLineEdit::keyPressEvent(event);

    qDebug()  <<" ChildLineEdit Accepted : " << event->isAccepted();

    event->ignore();

    if(event->key() == Qt::Key_Delete){
        qDebug() << "Pressed the Delete Key";
        clear();
    }else{
        ParentLineEdit::keyPressEvent(event);
    }
}

File: Chap1-Events/1-4EventPropagationDemo/childlineedit.py
from parentlineedit import ParentLineEdit
from PySide6.QtGui import QKeyEvent
from PySide6.QtCore import Qt

class ChildLineEdit(ParentLineEdit):
    def __init__(self, parent=None):
        super().__init__(parent)
    
    def keyPressEvent(self, event: QKeyEvent):
        print(f"ChildLineEdit Accepted: {event.isAccepted()}")
        
        event.ignore()
        
        if event.key() == Qt.Key.Key_Delete:
            print("Pressed the Delete Key")
            self.clear()
        else:
            super().keyPressEvent(event)